AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLSyncBased/GraphQLSyncCustomPrimaryKeyTests.swift

```diff
@@ -0,0 +1,219 @@
+@testable import AWSAPICategoryPluginTestCommon
+import AWSPluginsCore
+
+// swiftlint:disable cyclomatic_complexity
+class GraphQLSyncCustomPrimaryKeyTests: XCTestCase {
+
+    static let amplifyConfiguration = "GraphQLSyncBasedTests-amplifyconfiguration"
+
+    override func setUp() {
+        Amplify.reset()
+        let plugin = AWSAPIPlugin()
+
+        do {
+            try Amplify.add(plugin: plugin)
+
+            let amplifyConfig = try TestConfigHelper.retrieveAmplifyConfiguration(
+                forResource: GraphQLSyncBasedTests.amplifyConfiguration)
+            try Amplify.configure(amplifyConfig)
+
+            ModelRegistry.register(modelType: CustomerOrder.self)
+
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+    }
+
+    override func tearDown() {
+        Amplify.reset()
+    }
+
+    /// Test for deletion of model with custom primary keys
+    ///
+    /// - Given: A customer order
+    /// - When:
+    ///    - The customer order is queried for, updated, deleted and queried for again
+    /// - Then:
+    ///    - The customer order should be deleted
+    ///
+    func testDelete() throws {
+
+        /// Create order
+        let customerOrder = CustomerOrder(orderId: UUID().uuidString, email: "test@abc.com")
+        guard let mutationSyncResult = createCustomerOrder(customerOrder: customerOrder),
+              let _ = mutationSyncResult.model.instance as? CustomerOrder else {
+            XCTFail("Failed to create customer order")
+            return
+        }
+
+        /// Query order
+        let querySuccess = expectation(description: "query successful")
+        let queryRequest = queryCustomerOrder(modelName: CustomerOrder.modelName,
+                                         byId: customerOrder.id,
+                                         orderId: customerOrder.orderId)
+        Amplify.API.query(request: queryRequest) { result in
+            switch result {
+            case .success(let response):
+                switch response {
+                case .success(let mutationSync):
+                    guard let queriedCustomerOrder = mutationSync?.model.instance as? CustomerOrder else {
+                        XCTFail("Failed to retrieve customer order")
+                        return
+                    }
+                    XCTAssertEqual(queriedCustomerOrder.id, customerOrder.id)
+                    querySuccess.fulfill()
+                case .failure(let graphQLError):
+                    XCTFail("\(graphQLError)")
+                }
+            case .failure(let error):
+                XCTFail("\(error)")
+            }
+        }
+        wait(for: [querySuccess], timeout: TestCommonConstants.networkTimeout)
+
+        /// Update order
+        let updatedCustomerOrder = CustomerOrder(id: mutationSyncResult.model["id"] as? String ?? "",
+                                                 orderId: mutationSyncResult.model["orderId"] as? String ?? "",
```

Comment:
why check if `mutationSyncResult.model.instance` can be cast to CustomerOrder but then not use it?

Reply:
instead of creating a new object, could you try turning getting a var from line 55?
` var customerOrder = mutationSyncResult.model.instance as? CustomerOrder`

then here you could just update the email like `customerOrder.email = "updatedEmail" before passing it in